### DISCUSSION TITLE
Text Fields in Ribbons?

### DISCUSSION THREAD

**Discussion by MichaelWDenney** [kirill-grouchnikov/radiance#494] [Q&A] [answered]
Can a component extending a JTextField be added to a JRibbonBand or JFlowRibbonBand?

**Comment by kirill-grouchnikov** [comment, marked as answer]
Right now it's an "exercise for the reader". https://github.com/kirill-grouchnikov/radiance/tree/sunshine/component/src/main/java/org/pushingpixels/radiance/component/api/ribbon/synapse/projection has examples of how to extend core Swing components and expose them through the `ComponentProjection` and `JRibbonBand.addRibbonComponent` + `JFlowRibbonBand.addFlowComponent` APIs.
